examples/memstore-plugin/main.go

```diff
@@ -72,8 +72,9 @@ func main() {
 
 	memStorePlugin := grpcMemory.NewStoragePlugin(memory.NewStore(), memory.NewStore())
 	grpc.ServeWithGRPCServer(&shared.PluginServices{
-		Store:        memStorePlugin,
-		ArchiveStore: memStorePlugin,
+		Store:               memStorePlugin,
+		ArchiveStore:        memStorePlugin,
+		StreamingSpanWriter: memStorePlugin,
```

Comment:
iirc we're using this memstore-plugin for integration tests of the storage plugin framework, just like any other storage backend. However, in this implementation it looks like the streaming writes is the only write path that will be exercised, because you unconditionally switch to streaming writer if it's supported.

One way to address this is to run the tests another time with a different CLI flag such that in the first run `StreamingSpanWriter` field above will be left nil, and only populated if a special CLI flag is set.

Note that we already run gRPC test >1 times (`plugin/storage/integration/grpc_test.go`), to test both the hashicorp plugin and the regular remote storage.

Reply:
Got it, I set streamingWriter to be off by default and add a plugin flag `grpc-storage-plugin.write-type==streaming` to turn it on. Then add a new test method in `grpc_test.go.`